### PR TITLE
Remove misleading pluralization wording related to count metrics

### PR DESCRIPTION
**Pull request by trask**
Related to https://github.com/open-telemetry/semantic-conventions/pull/484#discussion_r1380203114

## Changes

Removes this text in the semconv:

> The pluralization rule does not apply in this case.

Because that text predated the "namespace pluralization rule" added in #267 and is meant to only apply to the other pluralization rules (not the namespace pluralization rule).

This is confusing now (with the addition of #267), and I think the easiest fix is just to remove the text, since the heading just above it already mentions it:

> #### Use `count` Instead of Pluralization for UpDownCounters

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* ~[schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.~

### CODE REVIEW

File: docs/general/metrics.md [hunk 1]
```diff
@@ -124,7 +124,6 @@ should be pluralized, even if only a single data point is recorded.
 
 If the value being recorded represents the count of concepts signified
 by the namespace then the metric should be named `count` (within its namespace).
-The pluralization rule does not apply in this case.
 
 For example if we have a namespace `system.processes` which contains all metrics related
```
Comment (joaopgrassi) [inline docs/general/metrics.md:128]: The namespace here in the example is still "pluralized". Is this correct? I find it still confusing.

File: CHANGELOG.md [hunk 2]
```diff
@@ -87,6 +87,10 @@ release.
 - Change the precedence between `:authority` and `Host` headers when populating
   `server.address` and `server.port` attributes.
   ([#455](https://github.com/open-telemetry/semantic-conventions/pull/455))
+- Clarify HTTP server definitions and `server.address|port` notes.
```
Comment (ChrsMark) [inline CHANGELOG.md:90]: Is this relevant?

File: CHANGELOG.md [hunk 3]
```diff
@@ -87,6 +87,10 @@ release.
...
+  ([#423](https://github.com/open-telemetry/semantic-conventions/pull/423))
```
Comment (trask) [inline CHANGELOG.md:91]:
```suggestion
```

File: docs/general/metrics.md (hunk 1 above)
Comment (trask) [reply docs/general/metrics.md]: updated

File: CHANGELOG.md (hunk 2 above)
Comment (trask) [reply CHANGELOG.md]: oops, fixed

**Review by AlexanderWert — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]

**Pull request merged**